### PR TITLE
login and logout

### PR DESCRIPTION
**Pull request by aravind289**

### CODE REVIEW

File: src/components/home/container.jsx [hunk 1]
```diff
@@ -17,9 +19,9 @@ class Fetch extends React.Component {
 
   componentDidMount() {
     getAllPackage()
-      .then(pkg_output => {
-        this.setState({ pkg: pkg_output, isloading: false })
-        console.log(pkg_output)
+      .then(packagese => {
```
Comment (digi0ps) [inline src/components/home/container.jsx]: Fix spelling mistake.

File: src/components/home/container.jsx [hunk 2]
```diff
@@ -42,25 +44,30 @@ class Fetch extends React.Component {
...
-          <div key={category_item._id}>
-            {' '}
-            <h2>
-              <Link to={`/category/${category_item._id}`}>
-                {category_item.name}
-              </Link>
-            </h2>
-          </div>
-        ))}
+        <Link to="/signup">Signup</Link>
```
Comment (digi0ps) [inline src/components/home/container.jsx]: Why isn't this inside a condition? i.e Where is welcome user?

File: src/components/login/loginpage.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+              name="password"
+              placeholder="password"
+              value={password}
+              onChange={this.changeHandler}
+            />
+          </div>
+
+          <div>
+            <button type="submit" onClick={this.submitHandler}>
+              Submit
```
Comment (digi0ps) [inline src/components/login/loginpage.jsx]: Write Login instead of Submit.

File: src/components/login/loginpage.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
+import { withRouter } from 'react-router-dom'
+import { PostCredentials } from '../../helpers/api'
+
+class LoginPage extends Component {
+  constructor(props) {
+    super(props)
+    //setiing initial state
+    this.state = {
+      email: 'abcde@gmail.com',
```
Comment (digi0ps) [inline src/components/login/loginpage.jsx]: Make state empty. This was just for testing.

File: src/components/login/signup.jsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+class SignUp extends Component {
+  constructor(props) {
+    super(props)
+    //setiing initial state
+    this.state = {
+      first_name: 'aravind',
+      last_name: 'krishnan',
+      email: 'abcde@gmail.com',
+      phone: '987654321',
```
Comment (digi0ps) [inline src/components/login/signup.jsx]: Same here, remove state values.

File: src/components/login/signup.jsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      email,
+      phone,
+      password,
+    }
+
+    PostUser(user).then(res => {
+      const { success } = res.data //deconstructing success
+
+      if ((status = 200 && success == true)) {
+        console.log('reg success')
```
Comment (digi0ps) [inline src/components/login/signup.jsx]: You should store entity in localstorage here too!

File: src/components/login/signup1.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+import { withRouter } from 'react-router-dom'
+
+import Home from '../home'
+import { PostUser } from '../../helpers/api'
+
+class Postform extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
```
Comment (digi0ps) [inline src/components/login/signup1.js]: Remove this file, not necessary.

File: src/components/router.jsx [hunk 8]
```diff
@@ -1,6 +1,8 @@
 import React from 'react'
 import { Switch, BrowserRouter as Router, Route } from 'react-router-dom'
 import Home from './home'
+import Signup from './login/signup1'
```
Comment (digi0ps) [inline src/components/router.jsx]: `login/signup`

File: src/helpers/api.js [hunk 9]
```diff
@@ -1,16 +1,34 @@
...
-  return fetch(`${BASE_Url}/package/${id}`).then(res => res.json())
+  return fetch(`${BASE_URL}/package/${id}`).then(res => res.json())
 }
 
 export function getCategoryId(id) {
-  return fetch(`${BASE_Url}/category/${id}`).then(res => res.json())
+  return fetch(`${BASE_URL}/category/${id}`).then(res => res.json())
+}
+
+export function PostUser(user) {
```
Comment (digi0ps) [inline src/helpers/api.js]: p small

File: src/helpers/api.js [hunk 10]
```diff
@@ -1,16 +1,34 @@
...
+  return axios({
+    method: 'POST',
+    url: `${BASE_URL}/user/register`,
+    data: user,
+  })
+}
+
+export function PostCredentials(user) {
+  return axios({
```
Comment (digi0ps) [inline src/helpers/api.js]: p small

File: src/helpers/api.js (hunk 9 above)
Comment (aravind289) [reply src/helpers/api.js]: y

Comment (digi0ps) [reply src/helpers/api.js]: Naming convention!

**Pull request merged**